server/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -614,15 +616,63 @@ public void messageReceived(StartedShardEntry request, TransportChannel channel,
                 request,
                 ClusterStateTaskConfig.build(Priority.URGENT),
                 shardStartedClusterStateTaskExecutor,
-                e -> {
-                    if (e instanceof FailedToCommitClusterStateException || e instanceof NotMasterException) {
-                        logger.debug(new ParameterizedMessage("failure during [shard-started {}]", request), e);
-                    } else {
-                        logger.error(new ParameterizedMessage("unexpected failure during [shard-started {}]", request), e);
+                new ClusterStateTaskListener() {
+                    @Override
+                    public void onFailure(Exception e) {
+                        logger.error(
```

Comment:
I think it'd be better to keep the `DEBUG` level for `FailedToCommitClusterStateException` and `NotMasterException`. `ERROR` is a bit overdramatic in any case here.

Reply:
++ made it debug in the `onNoLongerMaster` now.

Reply:
that fixes the `NotMasterException` case but not `FailedToCommitClusterStateException`

Reply:
:facepalm: right added conditional handling for `FailedToCommitClusterStateException` now as well.